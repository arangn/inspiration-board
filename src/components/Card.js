import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';

import './Card.css';

// class Card extends Component {
const Card = (props) => {


  // constructor(props) {
  //   super(props);
  // }

  // render() {
    return (
      <div className="card">
        <ul className="card__content">
          <li className="card__content-text">{props.text}</li>
          <li className="card__content-emoji">{emoji.getUnicode(`${props.emoji}`)}</li>
            <div>
              <button
                className="card__delete"
                onClick={() => props.deleteCardCallback(props.id)}>
                Delete Card
              </button>
            </div>
        </ul>
      </div>
    )
  }

Card.propTypes = {
  id: PropTypes.integer,
  text: PropTypes.string,
  emoji: PropTypes.string,
  deleteCardCallback: PropTypes.function
};

export default Card;
