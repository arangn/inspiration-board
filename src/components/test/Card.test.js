import React from 'react';
import Card from '../Card';
import { shallow } from 'enzyme';

describe('Card', () => {
  test('that it matches an existing snapshot', () => {
    // First Mount the Component in the testing DOM
    // Arrange
    const wrapper = shallow(
      <Card
        text="test name"
        emoji="test emoji"
        />
    );

    // Assert that it looks like the last snapshot
    expect(wrapper).toMatchSnapshot();
  });
});
