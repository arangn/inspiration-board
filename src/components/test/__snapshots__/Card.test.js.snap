// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card that it matches an existing snapshot 1`] = `
<div
  className="card"
>
  <ul
    className="card__content"
  >
    <li
      className="card__content-text"
    >
      test name
    </li>
    <li
      className="card__content-emoji"
    />
    <div>
      <button
        className="card__delete"
        onClick={[Function]}
      >
        Delete Card
      </button>
    </div>
  </ul>
</div>
`;
